cpp/modmesh/python/wrapper/modmesh/wrap_SimpleArray.cpp

```diff
@@ -157,6 +163,162 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapSimpleArray
             ;
     }
 
+    template <typename D /* for destination type */>
+    struct TypeBroadCast
+    {
+        static void broadcast(wrapped_type & arr_out, pybind11::array & arr_in)
+        {
+            using shape_type = small_vector<size_t>;
+            using out_type = typename std::remove_reference<decltype(arr_out[0])>::type;
+
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+            auto * arr_new = reinterpret_cast<pybind11::array_t<D> *>(&arr_in);
+
+            shape_type sidx_init(arr_out.ndim());
+
+            for (size_t i = 0; i < arr_out.ndim(); ++i)
+            {
+                sidx_init[i] = 0;
+            }
+
+            std::function<void(shape_type, int)> copy_idx;
+            copy_idx = [&](shape_type sidx, int dim)
+            {
+                if (dim < 0)
+                {
+                    return;
+                }
+
+                for (size_t i = 0; i < arr_out.shape(dim); ++i)
+                {
+                    sidx[dim] = i;
+
+                    size_t offset_in = 0;
+                    for (pybind11::ssize_t it = 0; it < arr_in.ndim(); ++it)
+                    {
+                        offset_in += arr_in.strides(it) / arr_in.itemsize() * sidx[it];
+                    }
+
+                    const D * ptr_in = arr_new->data();
+                    // NOLINTNEXTLINE
+                    arr_out.at(sidx) = static_cast<out_type>(*(ptr_in + offset_in));
+                    copy_idx(sidx, dim - 1);
+                }
+            };
+
+            copy_idx(sidx_init, arr_out.ndim() - 1);
+        }
+    };
+
+    // NOLINTNEXTLINE(readability-function-cognitive-complexity)
```

Comment:
I agree with clang-tidy.  This function is longer than 100 lines but it is not a complex tight loop.  Please follow tidy and break it.

Perhaps you can create a helper class like `TypeBroadCast`?